Neos.Flow/Classes/Reflection/ReflectionService.php

```diff
@@ -1224,7 +1224,15 @@ protected function reflectClass($className)
             throw new Exception\InvalidClassException('The class with name "' . $className . '" is a Doctrine proxy. It is not supported to reflect doctrine proxy classes.', 1314944681);
         }
 
-        $class = new ClassReflection($className);
+        // FIXME: This try catch was added with Flow 7.0 to improve the developer experience for upgrading components and can be removed later
```

Comment:
How about we rather build a separate flowpack for upgrade checks?

Reply:
The issue is that an existing class that still implements ComponentInterface will cause an exception during the compilation step (even during booting of the compilation step) already, where we don't have any hook-into mechanism in place. Otherwise I'd highly favor that of course

Reply:
Damn